core/src/View/Kokkos_ViewLegacy.hpp

```diff
@@ -236,13 +236,23 @@ class View : public ViewTraits<DataType, Properties...> {
  public:
   //----------------------------------------
   /** \brief  Compatible view of data type */
+#ifdef KOKKOS_ENABLE_IMPL_VIEW_LEGACY
+  using type = std::conditional_t<
+      has_hooks_policy,
+      View<typename traits::scalar_array_type, typename traits::array_layout,
+           typename traits::device_type, typename traits::hooks_policy,
+           typename traits::memory_traits>,
+      View<typename traits::scalar_array_type, typename traits::array_layout,
+           typename traits::device_type, typename traits::memory_traits>>;
+#else
```

Comment:
This change shouldn't be needed if `traits::data_type` resolves to `using data_type = typename data_analysis::type;` instead of `using data_type = typename data_analysis::data_type;` if `KOKKOS_ENABLE_IMPL_VIEW_LEGACY` is set. See your change in ViewTraits.hpp:L492

Reply:
hm, ok